### ISSUE TITLE
NT v2 50M fine-tuning script

### ISSUE THREAD

**Issue by smodee** [smodee/pathogen-classifier#11]
## Summary

Create `src/train_transformer.py` — fine-tune Nucleotide Transformer v2 50M (`InstaDeepAI/nucleotide-transformer-v2-50m-multi-species`) with HuggingFace Trainer.

**Priority: high | Depends on #9 (dataset.py)**
**Resolves:** #6 (class dominance via weighted loss)

## Implementation

Create `src/train_transformer.py` and `configs/train_nt.json`.

### Training strategy
- **Default: frozen encoder** — only train classification head (~600K trainable params)
- **Optional: unfreeze top N layers** via `--unfreeze-layers N` flag for better accuracy
- Standard HuggingFace training script — runs on Kaggle (T4), Azure ML (T4), or local CPU
- Device auto-detection (CUDA if available, else CPU)

### Development workflow
- **Kaggle (free T4):** All experimentation, hyperparameter tuning, ablations
- **Local CPU:** Quick debugging with `--max-train-samples 100`
- **Azure ML (Phase 3):** Final production training

### GPU training parameters (Kaggle T4 / Azure ML)
- `batch_size=16` (T4 has 16GB, NT v2 50M fits easily)
- `gradient_accumulation_steps=2` (effective batch 32)
- `fp16=True` when CUDA available
- `learning_rate=2e-4` for frozen encoder, `5e-5` if unfreezing layers
- `num_epochs=10` with early stopping (patience=3)

### CPU fallback parameters (local debugging)
- `batch_size=4`, `gradient_accumulation_steps=4`
- `fp16=False`
- `dataloader_num_workers=0` (Windows compatibility)

### Key classes/functions
- `WeightedTrainer(Trainer)` — overrides `compute_loss` with class-weighted CrossEntropyLoss (uses `compute_class_weights` from `src/dataset.py`)
- `compute_metrics(eval_pred)` — macro F1, weighted F1, balanced accuracy for Trainer callbacks
- `train(config_path)` — main entry point

### CLI
```bash
# Full training (on Kaggle/Azure with GPU)
python src/train_transformer.py --config configs/train_nt.json

# Quick debug run (local CPU)
python src/train_transformer.py --config configs/train_nt.json --max-train-samples 100
```

### Config: `configs/train_nt.json`
All hyperparameters: model name, batch sizes, learning rate, epochs, augmentation toggles, device settings.

### Model
`InstaDeepAI/nucleotide-transformer-v2-50m-multi-species` (55.9M params, 6-mer tokenization, 1000-token context)

## Dependencies from #9
- `SequenceDataset` from `src/dataset.py`
- `NTCollator` from `src/dataset.py`
- `compute_class_weights` from `src/dataset.py`
- `GenomicAugmentationDataset` from `src/augmentation.py`

## Acceptance criteria
- [ ] Training completes on CPU with `--max-train-samples 100` (smoke test)
- [ ] `WeightedTrainer` applies class weights correctly
- [ ] Augmentation pipeline integrates correctly during training
- [ ] Model checkpoints saved to `models/nt-v2/`
- [ ] Training metrics logged (loss, F1, balanced accuracy per epoch)
- [ ] Config file `configs/train_nt.json` is complete and documented
